app/controllers/api/v2/hosts_controller.rb

```diff
@@ -270,9 +270,10 @@ def power
 
       api :GET, '/hosts/:id/power', N_('Fetch the status of whether the host is powered on or not. Supported hosts are VMs and physical hosts with BMCs.')
       param :id, :identifier_dottable, required: true
+      param :timeout, String, required: false, desc: N_("Timeout to retrieve the power status of the host in seconds. Default is 3 seconds.")
 
       def power_status
-        render json: PowerManager::PowerStatus.new(host: @host).power_state
+        render json: PowerManager::PowerStatus.new(host: @host).power_state(params[:timeout])
```

Comment:
Unless `timeout` was specified in params, this will return `nil` and pass it to the `power_state` method explicitly. This will override the default `timeout = DEFAULT_TIMEOUT` variable :/

I guess it would be safer to rewrite it a bit:
```suggestion
        render json: PowerManager::PowerStatus.new(host: @host).power_state(params[:timeout] || DEFAULT_TIMEOUT)
```

Reply:
Nice catch @ofedoren !
I added a condition in the ping method itself, it should work as expected now.